crypto/fipsmodule/sha/internal.h

```diff
@@ -19,11 +19,16 @@
 extern "C" {
 #endif
 
-// SHA3_256 constants
+// SHA3 constants
 #define KECCAK1600_WIDTH 1600
 #define SHA3_256_CAPACITY_BYTES 64
 #define SHA3_256_DIGEST_BITLENGTH 256
 #define SHA3_256_DIGEST_LENGTH 32
+
+#define SHA3_512_CAPACITY_BYTES 128
```

Comment:
The Keccak state object only has space for:
```
uint8_t buf[KECCAK1600_WIDTH / 8 - SHA3_256_CAPACITY_BYTES];
```
`buf` just needs to have at least as many bytes needed by the algorithm with lowest capacity?

Can we statically assert somewhere that `buf` always has enough space?

Reply:
Thanks a lot for this point. Actually, in the case of only having SHA3-256 and SHA3-512, the lowest capacity is for SHA3-256. I will switch to `SHA3_MIN_CAPACITY_BYTES`, however, I can move back to `SHA3_224_CAPACITY_BYTES` later since this will be the lowest capacity if we add SHA3_224 and SHA3_384.